packages/modal/src/modalManager.ts

```diff
@@ -116,7 +116,12 @@ export class Web3Auth extends Web3AuthNoModal implements IWeb3AuthModal {
         },
       };
       if (!params?.modalConfig) params = { modalConfig: {} };
-      params.modalConfig = deepmerge(cloneDeep(params.modalConfig), adapterConfig);
+      const localSmsOtpEnabled = params.modalConfig[WALLET_ADAPTERS.AUTH]?.loginMethods?.[LOGIN_PROVIDER.SMS_PASSWORDLESS].showOnModal;
+      if (localSmsOtpEnabled === true && smsOtpEnabled === false) {
+        throw WalletInitializationError.invalidParams("must enable sms otp on dashboard in order to utilise it");
+      }
+
+      params.modalConfig = deepmerge(cloneDeep(adapterConfig), params.modalConfig);
```

Comment:
```suggestion
      params.modalConfig = deepmerge(adapterConfig, cloneDeep(params.modalConfig));
```
I think cloneDeep for adapter config is not required, as we just declared this above.